main.go

```diff
@@ -110,14 +155,153 @@ func runMain() error {
+		return nil
+	}
+
+	fmt.Println("Finished prererelease checks.")
+	
+	fmt.Println("Guessing next release version...")
+	latestReleaseVersion := getLatestReleaseVersion(repository, NO_PREVIOUS_VERSION)
+
+	// Conduct changelog file validation
+	changelogFilepath := path.Join(currentWorkingDirectory, relChangelogFilepath)
+	tbdHeaderCount := grepFile(changelogFilepath, TBD_VERSION_HEADER_REGEX)
+	if tbdHeaderCount != EXPECTED_NUM_TBD_HEADER_LINES {
+		fmt.Printf("There should be %d TBD header lines in the changelog. Instead there are %d.\n", EXPECTED_NUM_TBD_HEADER_LINES, tbdHeaderCount)
+		return nil
+	}
+	versionHeaderCount := grepFile(changelogFilepath, VERSION_HEADER_REGEX)
+	if versionHeaderCount == 0 {
+		fmt.Println("No previous changelog versions were detected in this changelog. Are you sure that the changelog is in sync with the release tags on this branch?")
+		return nil
+	}
+
+	existsBreakingChanges := detectBreakingChanges(changelogFilepath)
+	var nextReleaseVersion semver.Version
+	if existsBreakingChanges {
+		nextReleaseVersion = latestReleaseVersion.IncMinor()
+	} else {
+		nextReleaseVersion = latestReleaseVersion.IncPatch()
+	}
+
+	go wait()
+	fmt.Printf("VERIFICATION: Release new version '%s'? (ENTER to continue, Ctrl-C to quit))", nextReleaseVersion.String())
+	fmt.Scanln()
+
+	undoChanges := true
+	defer func() {
+		if undoChanges {
+			// git reset --hard origin/master
+			originMasterCommitHash, err := repository.ResolveRevision(plumbing.Revision(remoteMasterBranchName))
+			err = worktree.Reset(&git.ResetOptions{Mode: git.HardReset, Commit: *originMasterCommitHash })
+			if err != nil {
+				logrus.Errorf("ACTION REQUIRED: Error occurred attempting to undo local changes made for release '%s'. Please run 'git reset --hard origin/master' to undo manually.", nextReleaseVersion.String(), err)
+			}
+		}
+	}()
+
+	fmt.Println("Running prerelease scripts...")
+	preReleaseScriptsDirpath := path.Join(currentWorkingDirectory, relScriptsDirpath)
+	err = runPreReleaseScripts(preReleaseScriptsDirpath, nextReleaseVersion.String())
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while running prerelease scripts.")
+	}
+
+	fmt.Println("Updating the changelog...")
+	err = updateChangelog(changelogFilepath, nextReleaseVersion.String())
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while updating the changelog.")
+	}
+
+	fmt.Println("Committing changes locally...")
+	// Commit pre release changes
+	err = worktree.AddWithOptions(&git.AddOptions{All: true})
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while attempting to add release changes to staging area.")
+	}
+	commitMsg := fmt.Sprintf("Finalize changes for release version '%s'", nextReleaseVersion.String())
+	_, err = worktree.Commit(commitMsg, &git.CommitOptions{
+		Author: &object.Signature{
+			Name:  name,
```

Comment:
You'll want to pull `name` and `email` from the user's Gitconfig

Reply:
facts